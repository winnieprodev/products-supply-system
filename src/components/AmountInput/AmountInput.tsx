import React from "react";

import { IconButton } from "../Buttons";
import { MinusIcon, PlusIcon } from "../Icons";
import * as Styled from "./amountInput.styled";

interface OwnProps {
  value: number;
  maxLength?: number;
  onChange: (value: number) => void;
}

const AmountInput: React.FC<React.PropsWithChildren<OwnProps>> = ({
  value,
  onChange,
  maxLength = 4,
  children,
}) => {
  const increment = () => onChange(value + 1);
  const decrement = () => onChange(value - 1);

  const inputOnChange = (value: string) => {
    const numberValue = parseInt(value, 10);
    Number.isNaN(numberValue) ? onChange(0) : onChange(numberValue);
  };

  const valueForInput = value !== 0 ? value.toString() : "";

  return (
    <Styled.Wrapper>
      <IconButton primary onClick={decrement} disabled={value <= 0}>
        <MinusIcon />
      </IconButton>
      <Styled.Input
        value={valueForInput}
        placeholder="0"
        onChange={inputOnChange}
        maxLength={maxLength}
        onlyDigits
      />
      <IconButton primary onClick={increment}>
        <PlusIcon />
      </IconButton>
      <Styled.Description>{children}</Styled.Description>
    </Styled.Wrapper>
  );
};

export default React.memo(AmountInput);
