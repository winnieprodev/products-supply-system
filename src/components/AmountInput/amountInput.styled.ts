import styled from "styled-components";

import { Input as InputComponent } from "../Input";

export const Wrapper = styled.div`
  display: inline-flex;
  align-items: center;
`;

export const Description = styled.div`
  padding-left: 8px;
`;

export const Input = styled(InputComponent)`
  margin: 0 2px;
`;
