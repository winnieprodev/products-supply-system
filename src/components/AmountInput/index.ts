export { default as AmountInput } from "./AmountInput";
