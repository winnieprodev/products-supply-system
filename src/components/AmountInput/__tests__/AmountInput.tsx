import { render } from "@testing-library/react";

import AmountInput from "../AmountInput";

describe("AmountInput", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <AmountInput value={1} onChange={jest.fn()}>
          test
        </AmountInput>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for value=0 and maxLength specified", () => {
      const { container } = render(
        <AmountInput value={0} onChange={jest.fn()} maxLength={2}>
          test
        </AmountInput>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
