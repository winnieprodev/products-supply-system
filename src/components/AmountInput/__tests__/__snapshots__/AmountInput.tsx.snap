// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountInput Snapshot tests it matches the snapshot 1`] = `
.c1 {
  outline: none;
  cursor: pointer;
  border: 0 none;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  background-color: #F25C05;
  color: #FFFFFF;
}

.c1:hover {
  background-color: #DD5202;
}

.c1:active {
  background-color: #FF7626;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 22px;
  height: 22px;
  font-size: 10px;
  border-radius: 50%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  border: 0;
  font-size: 12px;
  background-color: #EEEEEE;
  color: #5F5F5F;
  border-radius: 6px;
  padding: 8px 12px;
  outline: none;
  box-shadow: inset 1px 2px 8px rgb(0,0,0,0.05);
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  width: 40px;
  text-align: right;
}

.c3::-webkit-input-placeholder {
  color: #bdbdbd;
}

.c3::-moz-placeholder {
  color: #bdbdbd;
}

.c3:-ms-input-placeholder {
  color: #bdbdbd;
}

.c3::placeholder {
  color: #bdbdbd;
}

.c3:hover {
  box-shadow: inset 2px 3px 8px rgb(0,0,0,0.1);
  background-color: #F3F3F3;
}

.c3:active {
  background-color: #E7E7E7;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  padding-left: 8px;
}

.c4 {
  margin: 0 2px;
}

<div
  class="c0"
>
  <button
    class="c1 c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-minus fa-w-14 "
      data-icon="minus"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        fill="currentColor"
      />
    </svg>
  </button>
  <input
    class="c3 c4"
    maxlength="4"
    placeholder="0"
    value="1"
  />
  <button
    class="c1 c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-plus fa-w-14 "
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        fill="currentColor"
      />
    </svg>
  </button>
  <div
    class="c5"
  >
    test
  </div>
</div>
`;

exports[`AmountInput Snapshot tests it matches the snapshot for value=0 and maxLength specified 1`] = `
.c1 {
  outline: none;
  cursor: pointer;
  border: 0 none;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  background-color: #F25C05;
  color: #FFFFFF;
  opacity: 0.3;
}

.c1:hover {
  background-color: #DD5202;
}

.c1:active {
  background-color: #FF7626;
}

.c5 {
  outline: none;
  cursor: pointer;
  border: 0 none;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  background-color: #F25C05;
  color: #FFFFFF;
}

.c5:hover {
  background-color: #DD5202;
}

.c5:active {
  background-color: #FF7626;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 22px;
  height: 22px;
  font-size: 10px;
  border-radius: 50%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  border: 0;
  font-size: 12px;
  background-color: #EEEEEE;
  color: #5F5F5F;
  border-radius: 6px;
  padding: 8px 12px;
  outline: none;
  box-shadow: inset 1px 2px 8px rgb(0,0,0,0.05);
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
  width: 20px;
  text-align: right;
}

.c3::-webkit-input-placeholder {
  color: #bdbdbd;
}

.c3::-moz-placeholder {
  color: #bdbdbd;
}

.c3:-ms-input-placeholder {
  color: #bdbdbd;
}

.c3::placeholder {
  color: #bdbdbd;
}

.c3:hover {
  box-shadow: inset 2px 3px 8px rgb(0,0,0,0.1);
  background-color: #F3F3F3;
}

.c3:active {
  background-color: #E7E7E7;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  padding-left: 8px;
}

.c4 {
  margin: 0 2px;
}

<div
  class="c0"
>
  <button
    class="c1 c2"
    disabled=""
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-minus fa-w-14 "
      data-icon="minus"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        fill="currentColor"
      />
    </svg>
  </button>
  <input
    class="c3 c4"
    maxlength="2"
    placeholder="0"
    value=""
  />
  <button
    class="c5 c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-plus fa-w-14 "
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        fill="currentColor"
      />
    </svg>
  </button>
  <div
    class="c6"
  >
    test
  </div>
</div>
`;
